WordPress/Classes/ViewRelated/Activity/ActivityListViewController.swift

```diff
@@ -104,3 +126,103 @@ extension ActivityListViewController: WPNoResultsViewDelegate {
         supportVC.showFromTabBar()
     }
 }
+
+// MARK: - ActivityRewindPresenter
+
+extension ActivityListViewController: ActivityRewindPresenter {
+
+    func presentRewindFor(activity: Activity) {
+        guard let rewindID = activity.rewindID,
+            !activity.isDiscarded && activity.rewindable else {
+            return
+        }
+
+        let title = NSLocalizedString("Rewind Site",
+                                      comment: "Title displayed in the Rewind Site alert, should match Calypso")
+        let rewindDate = activity.published.mediumStringWithUTCTime()
+        let message = NSLocalizedString("Are you sure you want to rewind your site back to \(rewindDate)\nThis will remove all content and options created or changed since then.",
+                                        comment: "Mesage displayed in the Rewind Site alert, the palceholder holds a date, should match Calypso.")
+
+        let alertController = UIAlertController(title: title,
+                                                message: message,
+                                                preferredStyle: .alert)
+        alertController.addCancelActionWithTitle(NSLocalizedString("Cancel", comment: ""))
+        alertController.addDestructiveActionWithTitle(NSLocalizedString("Confirm Rewind",
+                                                                        comment: "Confirm Rewind button title"),
+                                                      handler: { action in
+                                                        self.restoreSiteToRewindID(rewindID)
+                                                      })
+        self.present(alertController, animated: true, completion: nil)
+    }
+
+}
+
+// MARK: - Restores handling
+
+extension ActivityListViewController {
+
+    struct RestoreStatusCheck {
+        static let maxRetries = 12
+        static let pollingInterval = 5.0
+    }
+
+    fileprivate func restoreSiteToRewindID(_ rewindID: String) {
+        tableView.isUserInteractionEnabled = false
```

Comment:
I was going to suggest that SVProgressHUD already has blocking built in via `SVProgressHUD.setDefaultMaskType(.black)`, but that blocks the whole screen and the user can't event navigate away until the process is complete. I'm not sure if that's what we want, so your approach is probably better.

![simulator screen shot - iphone x - 2017-12-04 at 14 48 01](https://user-images.githubusercontent.com/8739/33555966-a2ef6ffc-d902-11e7-8cd1-54a92bf07da8.png)

Reply:
I'm sure once design is done with this one that hud will be gone to be honest.